### ISSUE TITLE
topic proposal: Redis-like key value store with write-ahead log

### ISSUE THREAD

**Issue by stuarth** [roc-lang/book-of-examples#32]
@gvwilson had suggested this, and I'd be very interested in writing it. That said, I don't believe it'd fit in an existing platform and would likely require something like a `basic-tcp`. Thoughts @rtfeldman or others?

**Comment by isaacvando** [comment]
This sounds very cool! Could the additional TCP primitives be added to basic-cli to accomplish it?

**Comment by Anton-4** [comment]
What would need to be added? [Current TCP features for reference](https://www.roc-lang.org/packages/basic-cli/0.8.1/Tcp).

**Comment by stuarth** [comment]
@Anton-4 we'd need to be be able to receive connections, a la `basic-webserver` but a TcpStream instead of an http request

**Comment by Anton-4** [comment]
Yeah, that seems like it would be out of scope for basic-cli. I would go with a dedicated tcp platform, I'll make a post about it on zulip to get some input.

**Comment by Anton-4** [comment]
https://roc.zulipchat.com/#narrow/stream/304641-ideas/topic/dedicated.20TCP.20platform/near/430409848

**Comment by gvwilson** [comment]
Can it be done as a pure library, i.e., skip the socket stuff and show how to build the data store? (that might help keep the size down)
